Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionCiscoMeraki.yaml

```diff
@@ -1,46 +1,580 @@
+                ),
+            EventMessage = iff(
+                    LogSubType has_any("Blocked DHCP server", "Virtual router collision"),
+                    Substring,
+                    coalesce(message, "")
+                ),
+            SrcUserName = column_ifexists("identity", ""),
+            SrcVlanId = coalesce(VLAN2, VLAN, vlan_id)
+        | extend
+            direction= iff(
+                direction has '"',
+                replace_string(tostring(direction), '"', ''),
+                replace_string(tostring(direction), "'", "")
+            ),
+            priority = iff(
+                priority has '"',
+                replace_string(tostring(priority), '"', ''),
+                replace_string(tostring(priority), "'", "")
+            ),
+            protocol = iff(
+                protocol has '"',
+                replace_string(tostring(protocol), '"', ''),
+                replace_string(tostring(protocol), "'", "")
+            ),
+            duration = iff(
+                duration has '"',
+                replace_string(tostring(duration), '"', ''),
+                replace_string(tostring(duration), "'", "")
+            ),
+            reason = iff(
+                reason has '"',
+                replace_string(tostring(reason), '"', ''),
+                replace_string(tostring(reason), "'", "")
+            ),
+            SrcVlanId =  iff(
+                    SrcVlanId has '"',
+                    replace_string(tostring(SrcVlanId), '"', ''),
+                    replace_string(tostring(SrcVlanId), "'", "")
+                ),
+            SrcUserName =  iff(
+                    SrcUserName has '"',
+                    replace_string(tostring(SrcUserName), '"', ''),
+                    replace_string(tostring(SrcUserName), "'", "")
+                )
+        | extend
+            EventResult = case(
+                    LogType in~("ids-alerts", "airmarshal_events", "security_event") or LogSubType == "association" or DvcAction == "Allow",
+                    "Success", 
+                    LogSubType == "disassociation" or LogSubType has_any ("Blocked DHCP server response", "Virtual router collision") or DvcAction == "Deny",
+                    "Failure",
+                    ""
+                ),
+            AdditionalFields = bag_pack(
+                        "signature",
+                        signature,
+                        "aid",
+                        aid,
+                        "rssi",
+                        rssi
+                    )
+        | invoke _ASIM_ResolveDvcFQDN('Device')
+        | lookup protocollookup on protocol
+        | lookup seveirtylookupforpriority on priority
+        | lookup severitylookupforDvcAction on DvcAction
+        | lookup directionlookup on direction
+        | lookup eventresultdetaillookupfordisassociation on reason
+        | parse kind=regex Substring with * "protocol=(icmp|icmp6) type="NetworkIcmpCode
```

Comment:
Regex can be avoided as follows:
 | parse Substring with * "protocol=" * " type=" NetworkIcmpCode

Reply:
Done
As we have divided it into code blocks, regex is removed.